src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantDetector.java

```diff
@@ -879,6 +852,153 @@ private static boolean alignmentIsDisjointFromAlphaOmega(final SimpleInterval al
+
+        EquivKey(final Kryo kryo, final Input input) {
+            final String ctg = input.readString();
+            final int start = input.readInt();
+            final int end = input.readInt();
+            affectedRefRegion = new SimpleInterval(ctg, start, end);
+
+            altSeq = new byte[input.readInt()];
+            input.read(altSeq);
+
+            segmentsAndAltArrange = serializer.read(kryo, input, ReferenceSegmentsAndEventDescription.class);
+        }
+
+        void serialize(final Kryo kryo, final Output output) {
+            output.writeString(affectedRefRegion.getContig());
+            output.writeInt(affectedRefRegion.getStart());
+            output.writeInt(affectedRefRegion.getEnd());
+            output.writeInt(altSeq.length);
+            output.write(altSeq);
+            serializer.write(kryo, output, segmentsAndAltArrange);
+        }
+
+        public static final class Serializer extends com.esotericsoftware.kryo.Serializer<EquivKey> {
+            @Override
+            public void write(final Kryo kryo, final Output output, final EquivKey x) {
+                x.serialize(kryo, output);
+            }
+
+            @Override
+            public EquivKey read(final Kryo kryo, final Input input,
+                                 final Class<EquivKey> clazz) {
+                return new EquivKey(kryo, input);
+            }
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            final EquivKey equivKey = (EquivKey) o;
+
+            if (!affectedRefRegion.equals(equivKey.affectedRefRegion)) return false;
+            if (!Arrays.equals(altSeq, equivKey.altSeq)) return false;
+            return segmentsAndAltArrange.equals(equivKey.segmentsAndAltArrange);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = affectedRefRegion.hashCode();
+            result = 31 * result + Arrays.hashCode(altSeq);
+            result = 31 * result + segmentsAndAltArrange.hashCode();
+            return result;
+        }
+
+        private Map<String, String> typeSpecificExtraAttributes() {
+            final Map<String, String> attributes = new HashMap<>();
+            attributes.put(CPX_EVENT_ALT_ARRANGEMENTS,
+                    segmentsAndAltArrange.getAltArrangementString());
+            if ( ! segmentsAndAltArrange.referenceSegments.isEmpty() ) {
+                attributes.put(CPX_SV_REF_SEGMENTS,
+                        String.join(VCFConstants.INFO_FIELD_ARRAY_SEPARATOR,
+                                segmentsAndAltArrange.referenceSegments.stream()
+                                        .map(SimpleInterval::toString).collect(Collectors.toList())));
+            }
+            return attributes;
+        }
+
+        VariantContext toVariantContext(final ReferenceMultiSource reference) throws IOException {
```

Comment:
What if you made this return a `VariantContextBuilder` instead of a `VariantContext`, then you wouldn't have to make it, then make a new builder from the new `VariantContext`, and then make that?